src/main/java/org/example/XMLValidator.java

```diff
@@ -0,0 +1,175 @@
+
+            int id = Integer.parseInt(userElement.getElementsByTagName("id").item(0).getTextContent());
+            String username = userElement.getElementsByTagName("username").item(0).getTextContent();
+            String email = userElement.getElementsByTagName("email").item(0).getTextContent();
+            String password = userElement.getElementsByTagName("password").item(0).getTextContent();
+            String registrationDate = userElement.getElementsByTagName("registration_date").item(0).getTextContent();
+
+            System.out.println("User ID: " + id);
+            System.out.println("Username: " + username);
+            System.out.println("Email: " + email);
+            System.out.println("Password: " + password);
+            System.out.println("Registration Date: " + registrationDate);
+            System.out.println();
+        }
+
+        Document reviewDoc = db.parse(reviewXMLFile);
+        NodeList reviewNodeList = reviewDoc.getElementsByTagName("review");
+
+        System.out.println("Parsing review xml");
+        System.out.println();
+        for (int i = 0; i < reviewNodeList.getLength(); i++) {
+
+            Node reviewNode = reviewNodeList.item(i);
+            Element reviewElement = (Element) reviewNode;
+
+            double rating = Double.parseDouble(reviewElement.getElementsByTagName("rating").item(0).getTextContent());
+            String comment = reviewElement.getElementsByTagName("comment").item(0).getTextContent();
+
+            System.out.println("Rating: " + rating);
+            System.out.println("Comment: " + comment);
+            System.out.println();
+        }
+
+        Document productDoc = db.parse(productXMLFile);
+        NodeList productNodeList = productDoc.getElementsByTagName("product");
+
+        System.out.println("Parsing product xml");
+        System.out.println();
+        for (int i = 0; i < productNodeList.getLength(); i++) {
+
+            Node productNode = productNodeList.item(i);
+            Element productElement = (Element) productNode;
+
+            String name = productElement.getElementsByTagName("name").item(0).getTextContent();
+            String description = productElement.getElementsByTagName("description").item(0).getTextContent();
+            double price = Double.parseDouble(productElement.getElementsByTagName("price").item(0).getTextContent());
+            int stockQuantity = Integer.parseInt(productElement.getElementsByTagName("stock_quantity").item(0).getTextContent());
+
+            System.out.println("Name: " + name);
+            System.out.println("Description: " + description);
+            System.out.println("Price: " + price);
+            System.out.println("Stock: " + stockQuantity);
+            System.out.println();
+        }
+
+        Document categoryDoc = db.parse(categoryXMLFile);
+        NodeList categoryNodeList = categoryDoc.getElementsByTagName("category");
+
+        System.out.println("Parsing category xml");
```

Comment:
logger would be better, same with the other ones like this

Reply:
fixed